### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by qubelka**
- Make on frontend the same checks as when creating a Reference model
- Prevent from submitting form without snames / references
- Do not update ref id on the sname (all snames will be connected to the one reference)

### CODE REVIEW

**Comment by tonimobin** [comment]
Everything seems to work, but an additional - more thorough review probably wouldn't hurt. Left a small question regarding doi's in slack.

File: app/frontend/src/components/Submit.js [hunk 1]
```diff
@@ -13,15 +14,33 @@ export const Submit = () => {
...
 
 	const submit = async () => {
+		if (data.reference && data.reference.edit) {
+			console.log(
+				'Please save changes made in reference before submitting'
+			)
+			return
+		}
+
+		if (data.structured_names.length === 0 || data.relations.length === 0) {
```
Comment (chzesa) [inline app/frontend/src/components/Submit.js:24]: Having no added structured names is probably a valid scenario if all new relations are between existing names.

File: app/frontend/src/validations.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		yearValidation: {
+			if (year) {
+				if (year === '0') {
+					console.log('Please provide correct year value')
+					valid = false
+					break yearValidation
+				}
+
+				const yearAsNumber = Number(year)
+				if (!yearAsNumber) {
```
Comment (chzesa) [inline app/frontend/src/validations.js:34]: The first conditional can be dropped if `yearAsNumber` is compared to NaN

File: app/frontend/src/validations.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+					break yearValidation
+				}
+
+				if (yearAsNumber < 1800 || yearAsNumber > 2100) {
+					console.log('Please provide correct year value')
+					valid = false
+				}
+			}
+		}
+	}
```
Comment (chzesa) [inline app/frontend/src/validations.js:46]: Year must be an integer to pass validation on server.

**Review by chzesa — CHANGES_REQUESTED** [review]

**Review by chzesa — APPROVED** [review]

**Pull request merged**